src/base/bittorrent/torrenthandleimpl.cpp

```diff
@@ -427,9 +427,15 @@ void TorrentHandleImpl::replaceTrackers(const QVector<TrackerEntry> &trackers)
         m_session->handleTorrentTrackersChanged(this);
     }
     else {
-        if (!currentTrackers.isEmpty())
+        if (!currentTrackers.isEmpty()) {
+            // Clear the peer list if it's a private torrent since
+            // we do not want to keep connecting with peers from old tracker.
+#if (LIBTORRENT_VERSION_NUM >= 10207)
+            if (this->isPrivate())
```

Comment:
Seems this is not correct, if `currentTrackers` is empty it means all existing trackers are being replaced/removed, and the clear peers action won't be executed.

I would prefer it is added to the bottom of this function (assuming it will still work as expected), such as:
```c++
void TorrentHandleImpl::replaceTrackers(const QVector<TrackerEntry> &trackers)
{
    if (newTrackers.isEmpty() && currentTrackers.isEmpty()) {
    }
    else {
    }
 
    // Clear the peer list if it's a private torrent since
    // we do not want to keep connecting with peers from old tracker.
    if (isPrivate())
        clearPeers();
}
```

Using `this` pointer is redundant, should be omitted.
Also libtorrent version guard isn't required in the current form.